src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.service.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.SaltUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        AuthenticationException exception =
+                new AuthenticationException("Login or password is incorrect");
+        User fondUser = userService.findByEmail(email).orElseThrow(() -> exception);
+        if (fondUser.getPassword().equals(SaltUtil.getSalt(password, fondUser.getSalt()))) {
+            return fondUser;
+        }
+        throw exception;
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (userService.findByEmail(email).isPresent()) {
+            throw new RegistrationException("User with this email already exists");
+        }
+        byte[] salt = new byte[2];
```

Comment:
Will be better if you will use your new method that randomly generates salt

Reply:
yes, first of all, we need to move the password hashing to the user service method as it is said in common mistakes, and we should use a HashUtil method, and bytes should be random and not hardcoded